### ISSUE TITLE
Empty strings with whitespaces and `EmptyValue` with quotes

### ISSUE THREAD

**Issue by EnricoMi** [uniVocity/univocity-parsers#353]
Writing "empty" strings that contain only whitespaces to CSV will quote-escape the `EmptyValue`. This does not happen when there are no whitespaces in the empty string.

Example:
`EmptyValue = "\"\""`
`IgnoreLeadingWhitespaces = true`
`IgnoreTrailingWhitespaces = true`

Empty strings will be written as `""`, whereas values with white-spaces only, e.g. `"\t"` or `" "` will first escape the quotes in the `EmptyValue` so they turn into `\"\"`.

Are quotes in the `EmptyValue` option discouraged, or should quotes in the `EmptyValue`not be quote-escaped by the writer, as I would imagine?

**Comment by jbax** [comment]
Can you provide an example with code that illustrates what you are not able to achieve?

Something with an expected output given some input would be great.

Thanks!

**Comment by EnricoMi** [comment]
Here is a unit test:

```
CsvWriterSettings s = new CsvWriterSettings();
CsvFormat format = s.getFormat();
format.setDelimiter(',');
format.setQuote('"');
format.setQuoteEscape('\\');
format.setComment('\u0000');

s.setIgnoreLeadingWhitespaces(true);
s.setIgnoreTrailingWhitespaces(true);
s.setNullValue("");
s.setEmptyValue("\"\"");
s.setSkipEmptyLines(true);
s.setQuoteAllFields(false);
s.setQuoteEscapingEnabled(true);
s.setErrorContentLength(1000);
CsvWriter w = new CsvWriter(s);

String result = w.writeRowToString(1, "\n");
assertEquals(result, "1,");
```

Result is:
```
java.lang.AssertionError: expected [1,] but found [1,\"\"]
Expected :1,
Actual   :1,\"\"
```

The parser cannot parse that CSV string:
```
CsvParserSettings s = new CsvParserSettings();
CsvFormat format = s.getFormat();
format.setDelimiter(',');
format.setQuote('"');
format.setQuoteEscape('\\');
format.setComment('\u0000');

s.setIgnoreLeadingWhitespaces(true);
s.setIgnoreTrailingWhitespaces(true);
s.setNullValue("");
s.setEmptyValue("\"\"");
s.setSkipEmptyLines(true);
s.setErrorContentLength(1000);
CsvParser parser = new CsvParser(s);

String[] result = parser.parseLine("1,\\\"\\\"");
assertEquals(result.length, 2);
assertEquals(result[0], "1");
assertEquals(result[1], "");
```

Result is:
```
java.lang.AssertionError: expected [] but found [\"\"]
Expected :
Actual   :\"\"
```

**Comment by jbax** [comment]
Thank you for the unit tests. It helps greatly.

For your last test the parser is reading this line correctly: `"1,\\\"\\\""`

Notice the quote escape is only processed if the value itself is inside quotes. In the case above, that value is NOT inside quotes, so you will get it back to you. If you want the parser to process escape sequences even though they are not quoted, add this:

`s.setEscapeUnquotedValues(true);`

Now, it will properly unescape these characters, and produce one quote followed by the other, i.e. `""` (this is a string of length = 2, just to avoid confusion).

It will never produce an empty string (i.e. with length 0) because the input contains quotes. They being escaped means that this is relevant content so it won't be ignored.

I'll check the first test now and get back to you soon.

**Comment by EnricoMi** [comment]
I am not concerned about the second test, it was merely to show that parsing the result of the writer does not reproduce the initial value (well, the trimmed version of the input string, hence the empty string).

**Comment by jbax** [comment]
Thank you! I've fixed the writing error on the latest 2.8.4-SNAPSHOT build.

Thank you for using our parsers!
